### ISSUE TITLE
[BUG] Cinnamon crashes in headless mode

### ISSUE THREAD

**Issue by DigiBC** [linuxmint/cinnamon#11142]
```
 * Cinnamon 5.4.11
 * Linux Mint 21
 * Intel Iris Plus Graphics 640 / i915 **not used** -> headless mode
 * Default display resolution (headless mode): 1024 x 768
 * 64 bit
 * Attached ~/.xsession-errors
 ```

### Issue
Linux Mint 21 running in headless mode without a monitor attached with VNC remote access (x11vnc / xrdp) crashes at startup after LightDM login.

### Steps to reproduce
If no monitor or no xorg.conf file with dummy monitor settings is present Cinnamon crashes.

Errors (.xsession-errors -> _LM21_headless_xsession-errors.txt_):
```
JS LOG: [LookingGlass/info] Monitor 0 not found. Not creating panel
(cinnamon:1420): Gjs-CRITICAL **: 13:44:09.191: JS ERROR: TypeError: layoutManager.primaryMonitor is undefined
start@/usr/share/cinnamon/js/ui/main.js:402:5
@<main>:1:47
```

### Expected behaviour
Earlier version of Cinnamon (≤ 5.2.7) in Linux Mint 20.x "ignored" a missing monitor (-> _LM20-3_headless_xsession-errors.txt_)
VNC remote access worked flawlessly.
Mint 21 Mate also works without any crashes.

### Other information
Use case: Several systems are running in headless mode controlled via VNC at a low display resolution (1024 x 768), but from time to time monitors with much higher resolutions are attached.
An xorg.conf file with dummy monitor settings doesn't work well in such a scenario. So, that's not a proper workaround. (And I'd prefer not to switch to the Mate desktop...)

[LM21_headless_xsession-errors.txt](https://github.com/linuxmint/cinnamon/files/9485046/LM21_headless_xsession-errors.txt)
[LM20-3_headless_xsession-errors.txt](https://github.com/linuxmint/cinnamon/files/9485047/LM20-3_headless_xsession-errors.txt)

**Comment by Matoran** [comment]
Same issue :+1:

**Comment by betops** [comment]
I upgraded a headless PC today and I run into the same issue, no idea how to fix it, I tried the xserver-xorg-video-dummy method as a workaround without luck...
Time to roll back now. Time to timeshift =)

**Comment by Matoran** [comment]
> I upgraded a headless PC today and I run into the same issue, no idea how to fix it, I tried the xserver-xorg-video-dummy method as a workaround without luck... Time to roll back now. Time to timeshift =)

xvfb works, but you will not have the same session from remote and local (if it's a full headless PC then it's not a problem).

**Comment by betops** [comment]
I think the easiest way to reproduce this issue is:

1. Fresh Install LM21 in a real pc/laptop, no VM
2. Configure autologin
3. Install and configure a remote desktop service like vnc server, xrdp, nomachine.
4. Power off the computer
5. Unplug the monitor so next time you boot up the computer no monitor is connected.
6. Power on the computer (the autologin will try and fail).
7. Try to start a remote session.
8. You probably see there is no panel nor window manager so, take a look at your .xsession-errors.

**Comment by mtwebster** [comment]
Ok I reproduced this.

To clarify, are you expecting to attach to the existing logged-in session, rather than starting a new one? If I disable autologin cinnamon starts fine.

**Comment by Matoran** [comment]
> Ok I reproduced this.
> 
> To clarify, are you expecting to attach to the existing logged-in session, rather than starting a new one? If I disable autologin cinnamon starts fine.

Yes I use it with x0vncserver to access to same session locally and remotely.

**Comment by betops** [comment]
> Ok I reproduced this.
> 
> To clarify, are you expecting to attach to the existing logged-in session, rather than starting a new one? If I disable autologin cinnamon starts fine.

yes, I am temporarily using xrdp without autologin starting a new session remotely, but with vnc or nomachine + autologin, it fails

**Comment by UlrichBoehm** [comment]
Any updates on this? I have same situation here with latest Linux Mint Cinnamon. Fresh installed, autologin. Disconnected monitor after everything was fine and did a reboot. Cinnamon never starts without connected monitor.
